File: java/fundamentals/day08/MiniChallenge.java
// File: java/fundamentals/day08/MiniChallenge.java
package fundamentals.day08;

import java.util.Arrays;

public class MiniChallenge {
    public static void main(String[] args) {
        int[] marks = {85, 72, 91, 68, 88};

        int total = 0;
        int highest = marks[0];
        int lowest = marks[0];
        int passed = 0;
        int failed = 0;
        int passMark = 35;

        for (int mark : marks) {
            total += mark;
            if (mark > highest) highest = mark;
            if (mark < lowest) lowest = mark;

            if (mark >= passMark) {
                passed++;
            } else {
                failed++;
            }
        }

        double average = (double) total / marks.length;
        String result = (failed == 0) ? "PASS" : "FAIL";

        // Create a sorted copy of the original array
        int[] sortedMarks = marks.clone();
        Arrays.sort(sortedMarks);

        // Display results
        System.out.println("==============================");
        System.out.println("     STUDENT MARKS ANALYZER");
        System.out.println("==============================");

        System.out.print("Marks       : ");
        for (int mark : marks) {
            System.out.print(mark + " ");
        }
        System.out.println();

        System.out.println("Total       : " + total);
        System.out.printf("Average     : %.2f%n", average);
        System.out.println("Highest     : " + highest);
        System.out.println("Lowest      : " + lowest);
        System.out.println("Passed      : " + passed);
        System.out.println("Failed      : " + failed);
        System.out.println("Result      : " + result);

        System.out.print("Sorted      : ");
        for (int mark : sortedMarks) {
            System.out.print(mark + " ");
        }
        System.out.println();
        System.out.println("==============================");
    }
}

File: python/fundamentals/day-08/MiniChallenge.py
# File: python/fundamentals/day-08/Mini_Challenge.py

def analyze_marks(marks):
    total = sum(marks)
    average = total / len(marks)
    highest = max(marks)
    lowest = min(marks)

    # Passing score threshold set to 35
    pass_mark = 35
    passed = sum(1 for mark in marks if mark >= pass_mark)
    failed = len(marks) - passed
    result = "PASS" if failed == 0 else "FAIL"

    sorted_marks = sorted(marks)

    # Output formatting
    print("==============================")
    print("     STUDENT MARKS ANALYZER")
    print("==============================")
    print(f"Marks       : {' '.join(map(str, marks))}")
    print(f"Total       : {total}")
    print(f"Average     : {average:.2f}")
    print(f"Highest     : {highest}")
    print(f"Lowest      : {lowest}")
    print(f"Passed      : {passed}")
    print(f"Failed      : {failed}")
    print(f"Result      : {result}")
    print(f"Sorted      : {' '.join(map(str, sorted_marks))}")
    print("==============================")


if __name__ == "__main__":
    marks = [85, 72, 91, 68, 88]
    analyze_marks(marks)